### DISCUSSION TITLE
Clickhouse http handlers

### DISCUSSION THREAD

**Discussion by jcolabre0514** [ClickHouse/ClickHouse#62833] [Q&A]
I am a newbie in using clickhouse, we defined 2 http handlers in config.xml and since the default http_port is 8123, all incoming http request should go via port 8123. After defining these 2 handlers, our spark jdbc connector that is used to connect and write data to clickhouse unable to with port 8123. Is there any other port that I can use for specifically for this spark jdbc connector? Or is there a proper way for me to define a handler that will allow this connections via spark jdbc? Need help. Thank you

**Comment by den-crane** [comment]
if you define your own http handler then you need to add a default handler as well. 
See the documentation https://clickhouse.com/docs/en/interfaces/http#predefined_query_handler

```
    <defaults/>     --------------<<<<<<<----------------
</http_handlers>
```

**Reply by jcolabre0514** [reply]
I have 2 handlers defined so does it mean I have to add <defaults/> on each handler?

**Reply by den-crane** [reply]
What do you mean? `<defaults/>` is under `http_handlers`

**Reply by jcolabre0514** [reply]
Hi Den, apolgies I understand wrongly initially but I got it to work. Anyway, got question after adding <defaults/> in http_handlers, does this mean that it will open up also normal SELECT queries from via API? Actually the main purpose of our http handlers is to hide the backgroud select queries from the external API application.
